overlord/servicestate/quota_handlers.go

```diff
@@ -595,7 +601,37 @@ func ensureSnapServicesStateForGroup(st *state.State, grp *quota.Group, opts *en
 	return restartSnapServices(st, nil, appsToRestartBySnap, nil)
 }
 
+func ensureGroupIsNotMixed(group string, allGrps map[string]*quota.Group) error {
+	grp, ok := allGrps[group]
+	if ok {
+		if len(grp.SubGroups) != 0 && len(grp.Snaps) != 0 {
+			return fmt.Errorf("quota group %q has mixed snaps and sub-groups, which is not supported anymore. please remove it and create it again to make any modifications", group)
+		}
+	}
+	return nil
+}
+
+func ensureGroupHasNoSubgroups(group string, allGrps map[string]*quota.Group) error {
+	grp, ok := allGrps[group]
+	if ok {
+		if len(grp.SubGroups) != 0 {
+			return fmt.Errorf("cannot mix snaps and sub groups in the group %q", group)
+		}
```

Comment:
nitpick: similar minor simplification, feel free to ignore

```suggestion
	if ok && len(grp.SubGroups) != 0 {
			return fmt.Errorf("cannot mix snaps and sub groups in the group %q", group)
```